src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,84 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH = 6;
+    private static RegistrationService registrationService;
+    private User userWithNotValidPassword;
+    private User userValid;
+    private User userWithInvalidAge;
+    private User userWithEqualLogin;
+    private User userWithNullLogin;
+    private User userWithNullPassword;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        userWithNullLogin = new User(null, "q3ff5e", 23);
+        userWithNullPassword = new User("jack", null, 43);
+        userWithNotValidPassword = new User("john", "49Gк5", 18);
+        userValid = new User("alice", "fOre8dfg", 35);
+        userWithInvalidAge = new User("rey", "dUrmd8sI", 15);

```

Comment:
```suggestion
        userWithInvalidAge = new User("rey", "dUrmd8sI", 15);
It's better to check critical values, for example, for age - 17 instead of 15.
```